File: ThePets/src/main/java/com/qualityplus/pets/base/commands/ReloadCommand.java
package com.qualityplus.pets.base.commands;

import com.qualityplus.assistant.TheAssistantPlugin;
import com.qualityplus.assistant.api.commands.command.AssistantCommand;
import com.qualityplus.assistant.lib.eu.okaeri.injector.annotation.Inject;
import com.qualityplus.assistant.util.StringUtils;
import com.qualityplus.pets.api.box.Box;
import com.qualityplus.pets.api.pet.Pets;
import com.qualityplus.pets.base.config.pet.Tiger;
import com.qualityplus.assistant.lib.eu.okaeri.commons.bukkit.time.MinecraftTimeEquivalent;

import com.qualityplus.assistant.lib.eu.okaeri.platform.bukkit.annotation.Delayed;
import com.qualityplus.assistant.lib.eu.okaeri.platform.core.annotation.Component;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import java.util.Collections;
import java.util.List;

@Component
public final class ReloadCommand extends AssistantCommand {
    private @Inject Box box;
    private @Inject Tiger tiger;

    @Override
    public boolean execute(CommandSender sender, String[] args) {
        Player player = (Player) sender;

        if (args.length == 1) {
            box.files().reloadFiles();

            Pets.reloadPets(tiger, box.plugin());

            player.sendMessage(StringUtils.color(box.files().messages().pluginMessages.successfullyReloaded));
        } else {
            player.sendMessage(StringUtils.color(box.files().messages().pluginMessages.useSyntax.replace("%usage%", syntax)));
        }
        return false;
    }

    @Override
    public List<String> onTabComplete(CommandSender commandSender, org.bukkit.command.Command command, String label, String[] args) {
        return Collections.emptyList();
    }

    @Delayed(time = MinecraftTimeEquivalent.SECOND)
    public void register(@Inject Box box) {
        TheAssistantPlugin.getAPI().getCommandProvider().registerCommand(this, e -> e.getCommand().setDetails(box.files().commands().reloadCommand));
    }
}

File: TheAnvil/src/main/java/com/qualityplus/anvil/base/commands/ReloadCommand.kt
package com.qualityplus.anvil.base.commands

import com.qualityplus.anvil.api.box.Box
import com.qualityplus.assistant.TheAssistantPlugin
import com.qualityplus.assistant.api.commands.command.AssistantCommand
import com.qualityplus.assistant.api.commands.setup.event.CommandSetupEvent
import com.qualityplus.assistant.util.StringUtils
import eu.okaeri.commons.bukkit.time.MinecraftTimeEquivalent
import eu.okaeri.injector.annotation.Inject
import eu.okaeri.platform.bukkit.annotation.Delayed
import eu.okaeri.platform.core.annotation.Component
import org.bukkit.command.Command
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

@Component
class ReloadCommand : AssistantCommand() {
    @Inject
    private val box: Box? = null
    override fun execute(sender: CommandSender, args: Array<String>): Boolean {
        val player = sender as Player
        if (args.size == 1) {
            box!!.getFiles()!!.reloadFiles()
            player.sendMessage(StringUtils.color(box.getFiles()!!.messages()!!.pluginMessages.successfullyReloaded))
        } else {
            player.sendMessage(
                StringUtils.color(
                    box!!.getFiles()!!.messages()!!.pluginMessages.useSyntax.replace("%usage%", syntax)
                )
            )
        }
        return false
    }

    override fun onTabComplete(
        commandSender: CommandSender,
        command: Command,
        label: String,
        args: Array<String>
    ): List<String> {
        return emptyList()
    }

    @Delayed(time = MinecraftTimeEquivalent.SECOND)
    fun register(@Inject box: Box) {
        TheAssistantPlugin.getAPI().commandProvider.registerCommand(this) { e: CommandSetupEvent<AssistantCommand> ->
            e.command.setDetails(
                box.getFiles()!!.commands()!!.reloadCommand
            )
        }
    }
}